$OpenBSD: patch-libswscale_swscale_c,v 1.2 2011/06/22 13:10:57 dcoppa Exp $

Fix a runtime CPU detection bug in libswscale that affected thumbnails
creation on amd64 (git commit 93c28a55fd84280d97c3c0dd7b0d546043242c34)

--- libswscale/swscale.c.orig	Wed Apr  6 01:28:59 2011
+++ libswscale/swscale.c	Tue Jun 21 14:17:21 2011
@@ -62,7 +62,6 @@ untested special converters
 #include "rgb2rgb.h"
 #include "libavutil/intreadwrite.h"
 #include "libavutil/x86_cpu.h"
-#include "libavutil/cpu.h"
 #include "libavutil/avutil.h"
 #include "libavutil/mathematics.h"
 #include "libavutil/bswap.h"
@@ -1315,12 +1314,6 @@ SwsFunc ff_getSwsFunc(SwsContext *c)
 {
 #if CONFIG_RUNTIME_CPUDETECT
     int flags = c->flags;
-
-    int cpuflags = av_get_cpu_flags();
-
-    flags |= (cpuflags & AV_CPU_FLAG_MMX ? SWS_CPU_CAPS_MMX : 0);
-    flags |= (cpuflags & AV_CPU_FLAG_MMX2 ? SWS_CPU_CAPS_MMX2 : 0);
-    flags |= (cpuflags & AV_CPU_FLAG_3DNOW ? SWS_CPU_CAPS_3DNOW : 0);
 
 #if ARCH_X86
     // ordered per speed fastest first
